src/FormField/Lookup.php

```diff
@@ -478,57 +412,46 @@ public function renderView()
-            $this->apiConfig = array_merge($this->apiConfig, ['data' => $this->getFilterQuery()]);
-
-            //render filters to template
-            $this->template->set('FilterHeaderLabel', $this->filterHeaderLabel);
-            $ft = $this->template->cloneRegion('Dropdown');
-            $html = '';
-
-            foreach ($this->filters as $k => $filter) {
-                $f_name = $this->name.'-ac_f'.$k;
-                $ft->set('input_id', $f_name);
-                $ft->set('FilterLabel', $filter['label']);
-                $ft->set('place_holder', $this->filterEmpty);
-                $ft->set('filterClass', 'atk-filter-dropdown');
-                $ft->setHTML('Input', $this->getFilterInput($filter['field'], $filter['field'].'_id'));
-                $html .= $ft->render();
-            }
-            $this->template->setHTML('Filters', $html);
-
-            // create proper js for dropdown.
-            $this->createFilterJsDropdown();
-
-            // output changed handler for filters.
-            $this->js(true, $this->onJsFilterChanged());
-        } else {
-            $this->template->del('FilterContainer');
-            $this->template->del('FilterLabelContainer');
+        if ($this->dependency) {
+            $this->apiConfig['data'] = array_merge([
+                'form' => new jsFunction([new jsExpression('return []', [$this->js()->closest('form')->serialize()])]),
+            ], $this->apiConfig['data'] ?? []);
         }
 
-        $this->js(true, $this->getJsDropdown());
+        $chain = new jQuery('#'.$this->name.'-ac');
+
+        $this->initDropdown($chain);
 
         if ($this->field && $this->field->get()) {
             $id_field = $this->id_field ?: $this->model->id_field;
-            $title_field = $this->title_field ?: $this->model->title_field;
 
             $this->model->tryLoadBy($id_field, $this->field->get());
 
-            if (!$this->model->loaded()) {
-                $this->field->set(null);
+            if ($this->model->loaded()) {
+                $row = $this->renderRow($this->model);
+
+                $chain->dropdown('set value', $row['value'])->dropdown('set text', $row['title']);
             } else {
-                $chain = new jQuery('#'.$this->name.'-ac');
-                // IMPORTANT: always convert data to string, otherwise numbers can be rounded by JS
-                $chain->dropdown('set value', (string) $this->model[$id_field])
-                        ->dropdown('set text', (string) $this->model[$title_field]);
-                $this->js(true, $chain);
+                $this->field->set(null);
             }
         }
 
+        $this->js(true, $chain);
+
         parent::renderView();
     }
+
+    /**
+     * Convert value to expected comma separated list before setting it.
+     *
+     * {@inheritdoc}
+     *
+     * @see \atk4\ui\FormField\Generic::set()
+     */
+    public function set($value = null, $junk = null)
```

Comment:
What is the use-case of this method? If we have `$this->values`, why do we need this?